### PR TITLE
remove ReportCollator message

### LINKED ISSUE #6415: Remove `ReportCollator` message
`ReportCollator` message is never sent by any subsystems. We should remove it along with this test

_Originally posted by @alindima in https://github.com/paritytech/polkadot-sdk/pull/4880#discussion_r1822129902_

### PR DESCRIPTION
**Pull request by jpserrat**
Closes #6415 

# Description

Remove unused message `ReportCollator` and test related to this message on the collator protocol validator side.

cc: @tdimitrov

### CODE REVIEW

**Review by bkchr — APPROVED** [review]

File: polkadot/node/subsystem-types/src/messages.rs
Comment (tdimitrov) [inline polkadot/node/subsystem-types/src/messages.rs:1]: Could you please remove `CollatorId` if it is not used anymore? I see a CI job complaining about it.

I can't comment on the exact line - it's 51, the error is shown on the PR (unused import).

**Comment by tdimitrov** [comment]
Good work @jpserrat! Thank you very much for the contribution.

Before merging we need a PRdoc for the pull request. It's a quick description of the changes in the PR which are used to release notes generation.

To create it add a file named pr_6628.prdoc (the ID of your PR) to the prdoc subdir in the root of the repo. There you need to explain your changes - what you wrote in the PR is more than enough. You can see an example here - https://github.com/paritytech/polkadot-sdk/blob/master/prdoc/pr_5198.prdoc

For your case audience will be `Node Dev` and you need to list the crates, changed by your PR (`polkadot-collator-protocol` and `polkadot-node-subsystem-types`).

File: polkadot/node/subsystem-types/src/messages.rs
Comment (jpserrat) [reply polkadot/node/subsystem-types/src/messages.rs:1]: Done!

**Comment by tdimitrov** [comment]
bot help

**Comment by tdimitrov** [comment]
bot clean

**Review by tdimitrov — APPROVED** [review]
Good work @jpserrat

**Pull request merged**